libipl/p10/ipl_sbe.C

```diff
@@ -0,0 +1,216 @@
+
+    rc = ipl_sbe_get_state(target, &state);
+
+    if (rc != 0)
+    {
+		ipl_log(IPL_ERROR, "Failed to read SBE state information (%s)", pdbg_target_path(pib));
+		return IPL_ERR_FSI_REG;
+	}
+
+	// SBE_STATE_CHECK_CFAM case is already handled by ipl_sbe_get_state function
+	if (state != SBE_STATE_BOOTED)
+    {
+		ipl_log(IPL_ERROR,
+			"SBE (%s) is not ready for chip-op: state(0x%08x)",
+			pdbg_target_path(pib), state);
+		return IPL_ERR_SBE_CHIPOP;
+	}
+
+	rc = sbe_mpipl_continue(target);
+
+	if (rc != 0)
+    {
+		ipl_log(IPL_ERROR, "SBE (%s) mpipl continue chip-op failed", pdbg_target_path(pib));
+		return IPL_ERR_SBE_CHIPOP;
+	}*/
+
+	return IPL_ERR_OK;
+}
+
+bool ipl_sbe_booted(TARGETING::TargetPtr target, uint32_t wait_time_seconds)
+{
+	sbeMsgReg_t sbeReg;
+	fapi2::ReturnCode fapi_rc;
+	uint32_t loopcount;
+
+	loopcount = wait_time_seconds > 0 ? wait_time_seconds : 25;
+
+	while (loopcount > 0)
+    {
+        std::cout << "p12-refactor executing p10_get_sbe_msg_register\n";
+	    fapi_rc = p10_get_sbe_msg_register(target, sbeReg);
+
+        if (fapi_rc == fapi2::FAPI2_RC_SUCCESS)
+        {
+			if (sbeReg.sbeBooted)
+            {
+                std::cout << "p12-refactor SBE Booted, wait time: " << loopcount << "\n";
+				ipl_log(IPL_INFO,
+					"SBE booted. sbeReg[0x%08x] Wait time: "
+					"[%d]\n",
+					uint32_t(sbeReg.reg), loopcount);
+				return true;
+			}
+            else
+            {
+                std::cout << "p12-refactor SBE boot is in progress. sbeReg[0x"
+                     << std::hex << std::setw(8) << std::setfill('0') << uint32_t(sbeReg.reg)
+                     << "]" << std::dec << std::endl;
+			}
+		}
+        else
+        {
+            std::cerr << "p12-refactor p10_get_sbe_msg_register failed fapi_rc = 0x"
+                << std::hex << static_cast<uint32_t>(fapi_rc)
+                << std::dec << std::endl;
+			
+            /*TODO ipl_log(IPL_ERROR,
+				"p10_get_sbe_msg_register failed for proc %d, "
+				"rc=%d\n",
+				pdbg_target_index(proc), fapi_rc);*/
+		}
+
+		loopcount--;
+		sleep(1);
+	}
+
+	// Get SBE debug data.
+	uint32_t val = 0xFFFFFFFF;
+
+    int rc = hwaccess::HwAccessIntf::getCfamRegister(target, 0x1007, val);
+
+    if (rc)
+    {
+		//TODO ipl_log(IPL_ERROR, "CFAM(0x1007) on %s failed", pdbg_target_path(fsi));
+	}
+
+    std::cout << "p12-refactor SBE Debug Data: 0x2809[0x"
+          << std::hex << std::setw(8) << std::setfill('0') << uint32_t(sbeReg.reg)
+          << "]  0x1007[0x"
+          << std::hex << std::setw(8) << std::setfill('0') << val
+          << "]" << std::dec << std::endl;
+    
+    return false;
+}
+
+int ipl_set_sbe_state_all(enum sbe_state state, bool skipMaster)
+{
+    using namespace TARGETING;
+    
+    int ret = 0;
+    auto& ts = TargetService::instance();
+
+    PredicateAttrVal<ATTR_TYPE> pred(TYPE_PROC);
+
+    auto top = ts.getTopLevelTarget();
+
+    for (auto&& proc :
```

Comment:
Please use predicate to simplify all these

    for (auto&& proc :
            ts.getAssociated(top, AssociationType::childByPhysical,
                             RecursionLevel::all, &pred))
    {
        if(skipMaster && ipl_is_master_proc(proc))
            continue;

		if (ipl_is_present(proc))